zio-http/src/main/scala/zio/http/netty/server/ServerInboundHandler.scala

```diff
@@ -264,13 +264,18 @@ private[zio] final case class ServerInboundHandler(
     runtime.run(ctx, ensured) {
       val pgm = for {
         response <- exit.sandbox.catchAll { error =>
-          ZIO.succeedNow {
-            error.failureOrCause
-              .fold[Response](
-                identity,
-                cause => HttpError.InternalServerError(cause = Some(FiberFailure(cause))).toResponse,
-              )
-          }
+          error.failureOrCause
+            .fold[UIO[Response]](
+              response => ZIO.succeedNow(response),
+              cause =>
+                (cause.dieOption match {
```

Comment:
Maybe the error callback should be a function `Cause[Nothing] => ZIO[Any, Nothing, Any]`?

This is also pushing a little on whether the exit should really be `ZIO[Any, Nothing, Response]`.

Reply:
Maybe. I think probably it's original goal was only to report netty exceptions, but I think the linked user request to also call it with ZIO die failures from handlers is reasonable. But I did not think of changing the callback type itself.

Reply:
Changed to `Cause[Nothing]`